### PR TITLE
Modify the dependency httpclient5 to httpclient4.

### PR DESCRIPTION
**Pull request by han-gao**
This PR is to resolve [this comment](https://github.com/Azure/azure-sdk-for-java/pull/16452#discussion_r508147974), remove httpclient5 dependency from jca.

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @han-gao , please:
1. Add description of this PR.
2. Please fix the pipeline failure.

**Comment by chenrujun** [comment]
Hi, @han-gao , please resolve the conflicts and fix the pipeline failure.

**Comment by han-gao** [comment]
I have tested the [sample](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/keyvault/azure-security-keyvault-jca/src/samples)and [Unit Test](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/keyvault/azure-security-keyvault-jca/src/test) under the [`jca`](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/keyvault/azure-security-keyvault-jca) module, the changed api will not affect the original function.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/LegacyRestClient.java [hunk 1]
```diff
@@ -60,20 +51,24 @@ public String post(String url, String body, String contentType) {
...
+            httpPost.setEntity(
+                new StringEntity(body, ContentType.create(contentType)));
+            result = client.execute(httpPost, responseHandler());
         } catch (IOException ioe) {
             ioe.printStackTrace();
         }
         return result;
     }
+
+    private ResponseHandler<String> responseHandler() {
```
Comment (yiliuTo) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/LegacyRestClient.java:63]: Shall we change the function name to "createResponseHandler"

Comment (han-gao) [reply sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/LegacyRestClient.java:63]: ok, it's easier to understand.

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**